### ISSUE TITLE
Build GLVis On mac BigSur (11.2)

### ISSUE THREAD

**Issue by zahurullah123** [mfem/mfem#2044]
I am getting the following error while building Glvis on Mac BigSur (11.2). I am following the instruction on https://mfem.org/building/. 

Is there any suggestion. 

```
Zahurs-MBP:mfem zahurullah$ cd glvis-4.0/
Zahurs-MBP:glvis-4.0 zahurullah$ make MFEM_DIR=../mfem-4.2 -j
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/gl/renderer.o -c lib/gl/renderer.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/gl/renderer_core.o -c lib/gl/renderer_core.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/gl/types.o -c lib/gl/types.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/aux_vis.o -c lib/aux_vis.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/font.o -c lib/font.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/openglvis.o -c lib/openglvis.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/palettes.o -c lib/palettes.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/sdl.o -c lib/sdl.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/vsdata.o -c lib/vsdata.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/vssolution.o -c lib/vssolution.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/vssolution3d.o -c lib/vssolution3d.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/vsvector.o -c lib/vsvector.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/vsvector3d.o -c lib/vsvector3d.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/gl/renderer_ff.o -c lib/gl/renderer_ff.cpp
g++ -O3 -std=c++11 -I../mfem-4.2 -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.0 -DGLVIS_OGL3 -I/opt/X11/include/freetype2 -DGLVIS_USE_LIBPNG -o lib/threads.o -c lib/threads.cpp
In file included from lib/palettes.cpp:13:
In file included from lib/aux_vis.hpp:15:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/gl/renderer_ff.cpp:12:
In file included from lib/gl/renderer_ff.hpp:15:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/gl/renderer.cpp:12:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/gl/types.cpp:12:
lib/gl/types.hpp:27:10: fatal error: 'glm/mat4x4.hpp' file not found
#include <glm/mat4x4.hpp>
         ^~~~~~~~~~~~~~~~
1 error generated.
1 error generated.
make: *** [lib/gl/types.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [lib/gl/renderer.o] Error 1
1 error generated.
make: *** [lib/gl/renderer_ff.o] Error 1
In file included from lib/openglvis.cpp:13:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/sdl.cpp:15:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
1 error generated.
In file included from lib/vsdata.cpp:20:
In file included from lib/vsdata.hpp:17:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
make: *** [lib/palettes.o] Error 1
1 error generated.
make: *** [lib/openglvis.o] Error 1
1 error generated.
make: *** [lib/sdl.o] Error 1
1 error generated.
make: *** [lib/vsdata.o] Error 1
In file included from lib/vssolution.cpp:21:
In file included from lib/visual.hpp:17:
In file included from lib/vsdata.hpp:17:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20In file included from :
lib/vsvector3d.cpp:19:
lib/gl/platform_gl.hppIn file included from :lib/visual.hpp15::1710:
:In file included from  lib/vsdata.hpp:17fatal error:
: In file included from lib/openglvis.hpp:16:
In file included from 'GL/glew.h'lib/sdl.hpp :file19 :
notIn file included from  lib/gl/renderer.hppfound:20
:
lib/gl/platform_gl.hpp:15:10: fatal error: #include <GL/glew.h>
         ^~~~~~~~~~~
'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/vssolution3d.cpp:19:
In file included from lib/visual.hpp:17:
In file included from lib/vsdata.hpp:17:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/aux_vis.cpp:20:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/vsvector.cpp:19:
In file included from lib/visual.hpp:17:
In file included from lib/vsdata.hpp:17:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
1 error generated.
1 error generated.
1 error generated.
make: *** [lib/vsvector.o] Error 1
make: *** [lib/vsvector3d.o] Error 1
make: *** [lib/vssolution.o] Error 1
1 error generated.
make: *** [lib/vssolution3d.o] Error 1
1 error generated.
make: *** [lib/aux_vis.o] Error 1
In file included from lib/threads.cpp:17:
In file included from lib/visual.hpp:17:
In file included from lib/vsdata.hpp:17:
In file included from lib/openglvis.hpp:16:
In file included from lib/sdl.hpp:19:
In file included from lib/gl/renderer.hpp:20:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
In file included from lib/gl/renderer_core.cpp:12:
In file included from lib/gl/attr_traits.hpp:15:
lib/gl/types.hpp:27:10: fatal error: 'glm/mat4x4.hpp' file not found
#include <glm/mat4x4.hpp>
         ^~~~~~~~~~~~~~~~
In file included from lib/font.cpp:14:
In file included from lib/font.hpp:22:
lib/gl/platform_gl.hpp:15:10: fatal error: 'GL/glew.h' file not found
#include <GL/glew.h>
         ^~~~~~~~~~~
1 error generated.
make: *** [lib/font.o] Error 1
1 error generated.
make: *** [lib/gl/renderer_core.o] Error 1
1 error generated.
make: *** [lib/threads.o] Error 1
```

**Comment by CODER591** [comment]
Hello, I am facing the same issue on the Ubuntu 18
It is simply lack of dependencies
My solution for Ubuntu:
Steps:
```
1. sudo apt-get install libglew-dev  (for the <GL/glew.h>)
2. sudo apt install libglm-dev
3. apt-get install libsdl2-dev
4. Re-run the build command.
```

**_For MacOS solution the same_**, you simply need to install these libraries as packages using `brew` or build & install them via their tutorials.

**Comment by zahurullah123** [comment]
Many Thanks, it worked with: 

```
brew install glew
brew install sdl2
brew install glm
```

**Comment by zahurullah123** [comment]
I installed GLVis but it is not working and gives this error, when I run ex2 in mfem: 

```
./glvis

       _/_/_/  _/      _/      _/  _/
    _/        _/      _/      _/        _/_/_/
   _/  _/_/  _/      _/      _/  _/  _/_/
  _/    _/  _/        _/  _/    _/      _/_/
   _/_/_/  _/_/_/_/    _/      _/  _/_/_/

Waiting for data on port 19916 ...
objc[2450]: +[NSApplication initialize] may have been in progress in another thread when fork() was called.
objc[2450]: +[NSApplication initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
```

**Comment by tzanio** [comment]
@zahurullah123, yes you need additional packages with the recent glvis-4.0 release, see also this note in the GLVis `INSTALL` file: https://github.com/GLVis/glvis/blob/master/INSTALL#L136-L138

@CODER591, thanks for helping answer this.

**Comment by tzanio** [comment]
> I installed GLVis but it is not working and gives this error, when I run ex2 in mfem:

Does `./glvis -m ../mfem/data/escher.mesh` work?

**Comment by tzanio** [comment]
Also, we are very open to PRs if you want to help us improve the documentation :)

**Comment by zahurullah123** [comment]
@tzanio  Yes this is worrking: 
`./glvis -m ../mfem/data/escher.mesh`

but when I use glvis this with e.g. exp2 it gives the above error.

**Comment by tzanio** [comment]
This could be related to this discussion: https://github.com/GLVis/glvis/discussions/141

Can you try the older version of SDL2 as described here: https://github.com/GLVis/glvis/discussions/141#discussioncomment-269943 ?

**Comment by zahurullah123** [comment]
@tzanio  yes, it was related to GLVis/glvis#141 it is working now. I just followed these steps: 

```
brew unlink sdl2
brew install ./sdl2.rb     (sdl2.rb.txt available from GLVis/glvis#141)
In glvis >>   make clean
In glvis >>   make MFEM_DIR=../mfem-4.2 -j
````

**Comment by tzanio** [comment]
Thanks for confirming @zahurullah123.

We are working on a more general solution to this issue in https://github.com/GLVis/glvis/pull/149
